Heavy-Division/B78XH
# fix: hotfix bug where APU shuts off when center pump is turned on

# APU/center pump bug fix

# Description

Bug from default Asobo 787 is fixed in which turning on the center fuel pumps, causes the APU to shut off. Did not matter if APU was already on, or if it was off and you wanted to turn it on - if center pumps were on, you have no APU. The workaround was simply to wait until after engine start to turn on the center fuel pumps.
(Interestingly some airlines actually do this, due to a bulletin where it was discovered that the center pumps had a low chance to fail during the electrical transfer from APU to engine power. However, that has nothing to do with the fuel feed of the APU, and the APU should absolutely be able to run with the center pumps on, and even receive center tank fuel.)

It was specifically the left center pump that would cause this issue - turning on only the right center pump would not cause problems. However, opening the crossfeed valve with the right center pump on would also cause the same issue.

The problem was that there are 2 fuel lines connected to the APU, and it was possible for each to feed the APU with different pressures. The line with stronger pressure (which is from center pump) would override the other, and send fuel flow (and pressure) backwards across the other line. Unfortunately, the logic behind the fuel system components encoded in flight_model.cfg is not known to me, so I can only guess at why it behaves like this. However my theory is that when the APU senses fuel going out from any direction, it glitches out and shuts down, and the APU pump shuts off as well.

This was resolved by redirecting the lines so that both can still send fuel to the APU, but there is only 1 line feeding it directly. Thus it's no longer possible for there to be any outward fuel pressure from the APU, and it no longer shuts off unexpectedly. This is similar to engine fuel feed, where there is only 1 path where fuel can be supplied, so the engines can never run into this same issue.


# Source material

From asking 787 pilots about RL behavior


# Testing Instructions 

Go through normal procedures with fuel in the center tank. Turn on all fuel pumps and then turn on the APU - it should turn on. Also try turning on the APU with only left and/or right tank pumps on, and after the APU has started up, turn the center pumps on - APU should stay on.

# Test Results




# Discord Username

oz#1558
